packages/react/src/use-chat.ui.test.tsx

```diff
@@ -556,3 +563,303 @@ describe('maxToolRoundtrips', () => {
+
+        <form
+          onSubmit={event => {
+            handleSubmit(event, {
+              experimental_attachments: attachments,
+            });
+            setAttachments(undefined);
+            if (fileInputRef.current) {
+              fileInputRef.current.value = '';
+            }
+          }}
+          data-testid="chat-form"
+        >
+          <input
+            type="file"
+            onChange={event => {
+              if (event.target.files) {
+                setAttachments(event.target.files);
+              }
+            }}
+            multiple
+            ref={fileInputRef}
+            data-testid="file-input"
+          />
+          <input
+            value={input}
+            onChange={handleInputChange}
+            disabled={isLoading}
+            data-testid="message-input"
+          />
+          <button type="submit" data-testid="submit-button">
+            Send
+          </button>
+        </form>
+      </div>
+    );
+  };
+
+  beforeEach(() => {
+    render(<TestComponent />);
+  });
+
+  afterEach(() => {
+    vi.restoreAllMocks();
+    cleanup();
+  });
+
+  it('should handle text file attachment and submission', async () => {
+    const file = new File(['test file content'], 'test.txt', {
+      type: 'text/plain',
+    });
+
+    const { requestBody } = mockFetchDataStream({
+      url: '/api/stream-chat',
+      chunks: ['0:"Response with attachment"\n'],
+    });
+
+    const fileInput = screen.getByTestId('file-input');
+    await userEvent.upload(fileInput, file);
+
+    const messageInput = screen.getByTestId('message-input');
+    await userEvent.type(messageInput, 'Message with attachment');
+
+    const submitButton = screen.getByTestId('submit-button');
+    await userEvent.click(submitButton);
+
+    const sentBody = JSON.parse((await requestBody) as string);
+    expect(sentBody.messages[0].content).toBe('Message with attachment');
+    expect(sentBody.messages[0].experimental_attachments).toBeDefined();
+    expect(sentBody.messages[0].experimental_attachments.length).toBe(1);
+    expect(sentBody.messages[0].experimental_attachments[0].name).toBe(
+      'test.txt',
+    );
+
+    await screen.findByTestId('message-0');
+    expect(screen.getByTestId('message-0')).toHaveTextContent(
+      'User: Message with attachment',
+    );
+
+    await screen.findByTestId('attachment-0');
+    expect(screen.getByTestId('attachment-0')).toHaveTextContent(
+      'test file content',
+    );
+
+    await screen.findByTestId('message-1');
+    expect(screen.getByTestId('message-1')).toHaveTextContent(
+      'AI: Response with attachment',
+    );
+  });
+
+  // image file
+
+  it('should handle image file attachment and submission', async () => {
+    const file = new File(['test image content'], 'test.png', {
+      type: 'image/png',
+    });
+
+    const { requestBody } = mockFetchDataStream({
+      url: '/api/stream-chat',
+      chunks: ['0:"Response with image attachment"\n'],
```

Comment:
`response to message with image attachment` ? (it confused me the first time i saw it)

Reply:
done!